OpenRA.Game/Network/OrderManager.cs

```diff
@@ -146,27 +146,24 @@ public void TickImmediate()
 						CheckSync(packet);
 					}
 					else if (frame == 0)
-						immediatePackets.Add((clientId, packet));
+					{
+						foreach (var o in packet.ToOrderList(World))
+						{
+							UnitOrders.ProcessOrder(this, World, clientId, o);
+
+							// A mod switch or other event has pulled the ground from beneath us
+							if (disposed)
```

Comment:
Wouldn't it make more sense to put this before the foreach loop, or even change the `else if` to
`else if (frame == 0 && !disposed)`?

Reply:
Good catch. This check should move up to the top since we don't want to be processing *any* new orders once we've been disposed.